### PR TITLE
feat: Add new component for detour route selection panel

### PR DESCRIPTION
**Pull request by joshlarson**

### CODE REVIEW

**Comment by firestack** [comment]
just noting this, I'm reviewing this by using it!

**Review by firestack — COMMENTED** [review]
I'm about ready to approve this, just thinking through a few things.

File: assets/src/components/detours/detourRouteSelectionPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+type SelectedRouteInfo =
+  | SelectedRouteInfoWithRoute
+  | SelectedRouteInfoWithoutRoute
+
+interface DetourRouteSelectionPanelProps {
+  allRoutes: Route[]
+  selectedRouteInfo: SelectedRouteInfo
+}
+
+const selectedRoutePatternForInfo = (
```
Comment (firestack) [inline assets/src/components/detours/detourRouteSelectionPanel.tsx:31]: super minor non-blocking nit: "for" sounds weird to me and I'd _personally_ prefer `From` but don't feel pressured to.
```suggestion
const selectedRoutePatternFromInfo = (
```

Comment (joshlarson) [reply assets/src/components/detours/detourRouteSelectionPanel.tsx]: [I also like that better](https://github.com/mbta/skate/pull/2614/commits/f5fb48751652ed56039441d4700a82ae5a774f33)

File: assets/src/components/detours/detourRouteSelectionPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            {allRoutes.map((route) => (
+              <option
+                key={route.id}
+                value={route.id}
+                selected={route.id === selectedRouteInfo.selectedRoute?.id}
+              >
+                {route.name}
+              </option>
+            ))}
+          </FormSelect>
```
[the remark this answers is not in the document]
Comment (joshlarson) [reply assets/src/components/detours/detourRouteSelectionPanel.tsx]: Thanks for catching that!

**Review by firestack — APPROVED** [review]

**Pull request merged**